tests/containers/multi_runtime.pm

```diff
@@ -0,0 +1,118 @@
+# SUSE's openQA tests
+#
+# Copyright 2024-2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Validate docker and podman network interoperability
+# - validate if podman and docker networking works together
+# - check if firewalld doesn't break either
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use Mojo::Base qw(containers::basetest);
+use testapi;
+use serial_terminal qw(select_serial_terminal select_user_serial_terminal);
+use utils;
+use containers::common qw(install_packages);
+use Utils::Logging 'save_and_upload_log';
+
+my @containers;
+my $test_image = "registry.opensuse.org/opensuse/nginx";
+
+sub run_tests {
+    my $test_url = "http://www.opensuse.org";
+    my $volumes = '-v $HOME/nginx/nginx.conf:/etc/nginx/nginx.conf:ro,z -v $HOME/nginx:/usr/share/nginx/html:ro,z';
+    my $curl_opts = "-svL -o /dev/null";
+    my $port = 8000;
+
+    # Create rootfull & rootless containers on both podman & docker
+    for my $runtime ("podman", "docker") {
+        for my $type ("root", "rootless") {
+            my $name = "test_${type}_${runtime}";
+            my $sudo = ($type eq "root") ? "sudo" : "";
+            script_retry("$sudo $runtime pull $test_image", retry => 3, delay => 60, timeout => 180);
+            assert_script_run "$sudo $runtime run -d --name $name -p $port:80 $volumes $test_image";
+            push @containers, {name => $name, runtime => $runtime, sudo => $sudo, port => $port};
+            $port++;
+        }
+    }
+
+    # Test connectivity to all containers on localhost
+    for my $container (@containers) {
+        assert_script_run "curl -4 $curl_opts http://127.0.0.1:$container->{port}",
+          fail_message => "failed IPv4 localhost test for $container->{name}";
+    }
+
+    # Test container connectivity to the outside world
+    for my $container (@containers) {
+        assert_script_run "$container->{sudo} $container->{runtime} exec $container->{name} curl -4 $curl_opts $test_url",
+          fail_message => "failed IPv4 Internet test for $container->{name}";
+    }
+}
+
+sub run {
+    select_serial_terminal;
+
+    my @pkgs = ("docker", "podman");
+    install_packages(@pkgs);
```

Comment:
For better compatibility we should utilize the `install_podman_when_needed` and `install_docker_when_needed` routines from `lib/containers/common.pm` instead.

Reply:
> For better compatibility we should utilize the `install_podman_when_needed` and `install_docker_when_needed` routines from `lib/containers/common.pm` instead.

Those routines do a lot more than just installing the packages and make some assumptions that I prefer to avoid here.

Reply:
But they are meant to work in all conditions, so if you say they don't work for you here, then those routines should be adapted such that they do.

Reply:
I'd rather make an exception for this module given that those functions mess with the firewall configuration, which is something that we want to control here.  Adding parameters to avoid this would be messy.  Also, public cloud tests install docker & podman on their own.